alchemy-service/src/test/java/io/rtr/alchemy/service/resources/ExperimentsResourceTest.java

```diff
@@ -77,6 +77,36 @@ public void testAddExperiment() {
         assertNotNull(experiment("new_experiment"));
     }
 
+    @Test
+    public void testAddDuplicateExperiment() {
+        final CreateExperimentRequest request =
+                new CreateExperimentRequest(
+                        "duplicate_experiment_name",
+                        0,
+                        "it will be a duplicate",
+                        null,
+                        null,
+                        false,
+                        Lists.newArrayList(
+                                new TreatmentDto("control", "the default"),
+                                new TreatmentDto("first_case", "the first case")
+                        ),
+                        Lists.newArrayList(
+                                new AllocateRequest("control", 50),
+                                new AllocateRequest("first_case", 50)
+                        ),
+                        Lists.<TreatmentOverrideRequest>newArrayList()
+                );
+
+        put(EXPERIMENTS_ENDPOINT)
+                .entity(request)
+                .assertStatus(Status.CREATED);
+        assertNotNull(experiment("duplicate_experiment_name"));
+        put(EXPERIMENTS_ENDPOINT)
+                .entity(request)
+                .assertStatus(Status.INTERNAL_SERVER_ERROR);
```

Comment:
The exception and 400 is getting swallowed somewhere in the tests and instead a 500 is returned.